compiler/rustc_codegen_ssa/src/mir/constant.rs

```diff
@@ -67,6 +67,12 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
     ) -> Result<Option<ty::ValTree<'tcx>>, ErrorHandled> {
         let uv = match constant.literal {
             mir::ConstantKind::Unevaluated(uv, _) => uv.shrink(),
+            mir::ConstantKind::Ty(c) => match c.kind() {
+                // A constant that came from a const generic but was then used as an argument to old-style
+                // simd_shuffle (passing as argument instead of as a generic param).
+                rustc_type_ir::ConstKind::Value(valtree) => return Ok(Some(valtree)),
+                other => span_bug!(constant.span, "{other:#?}"),
+            },
             other => span_bug!(constant.span, "{other:#?}"),
```

Comment:
Could you also handle `ConstantKind::Val`?
The fact that const-prop does not replace `Unevaluated` by `Val` in this very case is not documented, and may change by accident.

Reply:
I specifically do not want to handle it. I removed the logic that allows destructuring ConstValue except for pretty printing. I am currently working on eliminating this simd_shuffle special case entirely by making it a real const generic. That should avoid any such issues

Reply:
I left a comment about this.